---
published: true
title: "Detection of cancer signatures by integrating different features from omics data combining multi-block and path modelling methods"
layout: post
date: 2014-03-27 00:00
projects: true
hidden: true # don't count this post in blog pagination
category: poster
author: carleshf
externalLink: false
---



## Detection of cancer signatures by integrating different features from omics data combining multi-block and path modelling methods

___Juan R. Gonzalez,___ _Mikel Esnaola, Alejando Caceres, Carles Hernandez-Ferrer, Marcos Lopez-Sanchez Luis A. Perez-Jurado_

### Integrative and Computational Biology Joint Symposium 

 - _Date_: 27 and 28 of March 2014
 - _[Program]({{baseurl}}/assets/integrative_and_computational_biology_joint_symposium.pdf)_

### Abstract

The main goal of personalized medicine is to use medical models to customize health care by clustering individuals based on particular profiles. The use of ‘omic’ data has played a major role in these profiling efforts. Data integration procedures are required before performing such clustering methods. However, data integration has to face the challenge of dealing with the huge dimensionality of multi-modal data while maximizing their individual information. Several techniques have been applied to reduce the dimension of the space of predictors in ‘omic’ data, but none of them guarantees the interpretability of the results, or even biological sense of the latent variables. To address this limitation, we propose to reduce the dimensionality by extracting the relevant features from each ‘omic’ data. For instance, genomic data can provide additional information about gene-sets, point mutations, copy number variants, inversions and mosaic events. Alternatively, RNA-seq data can inform about global transcriptomic expression and alternative splicing events. Once data are summarized, we use a novel approach to perform data integration called regularized generalized canonical correlation. This method incorporates two different ways of combining datasets. Fist, it provides a consensus between tables (e.g. similar to PCA for more than two tables). Second, it includes information about how datasets are linked (i.e. data at different time-points, causality relationships,...). Outputs are then used to: 1) describe profiles of individuals having similar phenotypes to define “disease signatures”, and 2) determine those individuals with similar patterns which may help to stratify the population or describe new phenotypes. Initial results include analysis on data from the TCGA (The Cancer Genome Atlas) project including genomic, epigenomic, transcriptomic and clinical information, to study the determinants of cancer prognosis. Our approach finds biomarkers that correlate with complex combinations of different features obtained from ‘omic’ data that improves individuals’ prognosis prediction.