---
published: true
title: "WEScover"
layout: post
date: 2018-07-18 22:10
projects: true
hidden: true # don't count this post in blog pagination
description: "Shiny web application to explore the breadth of coverage per gene/exon obtained using WES technology in 1KGP."
category: project
author: carleshf
externalLink: false
image: https://carleshf.github.io/assets/portraits/gnomAD_notch1.png
headerImage: true
---


## Summary

WEScover is a R/Shiny web application to highlight global breadth of coverage in gene level and inter-individual variation for genes along with corresponding genetic tests listed in the National Institutes of Health Genetic Testing Registry (GTR). WEScover helps to check whether genes of interest could be sufficiently covered in terms of breadth and depth by whole exome sequencing (WES). With the goal to minimize the change of false negative, for each transcript, breadth of coverage data was calculated at 10x, 20x, and 30x read depth from the 1000 Genomes Project (1KGP).

## Results

The application is now running:

* WEScover: [link](http://gnome.tchlab.org/WEScover/)

The manuscript describing the tool and its aims is currently in **bioRxiv**:

* bioRxiv: [link](https://www.biorxiv.org/content/early/2018/07/27/367607)
